cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocProvisioner.java

```diff
@@ -337,13 +340,24 @@ public Capabilities getCapabilities() {
     }
 
     // Default settings from the system context
-    List<String> keys = Arrays.asList(DataprocConf.PREFER_EXTERNAL_IP,
-                                      DataprocConf.NETWORK,
-                                      DataprocConf.NETWORK_HOST_PROJECT_ID,
-                                      DataprocConf.STACKDRIVER_LOGGING_ENABLED,
-                                      DataprocConf.STACKDRIVER_MONITORING_ENABLED,
-                                      DataprocConf.IMAGE_VERSION,
-                                      BUCKET, RUNTIME_JOB_MANAGER);
+    ArrayList<String> keys = new ArrayList<>(
+      Arrays.asList(DataprocConf.PREFER_EXTERNAL_IP,
+                    DataprocConf.NETWORK,
+                    DataprocConf.NETWORK_HOST_PROJECT_ID,
+                    DataprocConf.STACKDRIVER_LOGGING_ENABLED,
+                    DataprocConf.STACKDRIVER_MONITORING_ENABLED,
+                    DataprocConf.IMAGE_VERSION,
+                    BUCKET, RUNTIME_JOB_MANAGER)
+    );
+
+    // Default cluster properties (i.e. specifying configuration files of the dataproc cluster and the property
+    // key value paris within those files that should be overridden) from the system context.
+    List<String> clusterPropertyKeys = Collections.unmodifiableList(
```

Comment:
Seems like calling `unmodifiableList` is unnecessary.

Reply:
Done. Changed to appending to list directly